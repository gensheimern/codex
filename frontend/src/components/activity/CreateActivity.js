import React from "react";
import {
    Button,Alert,Checkbox,
    ButtonGroup,Popover,Tooltip,Modal,OverlayTrigger,Navbar,Nav,NavItem,MenuItem,NavDropdown,FormGroup,FormControl
} from "react-bootstrap";
import 'react-bootstrap-table-next/dist/react-bootstrap-table2.min.css';
import config from '../../config';
import logo from '../../IMG/codex_logo1x.png';

export default class CreateActivity extends React.Component {
  constructor(props) {
      super(props);
      this.state = {
          groups: [],
          show: false,
          description: "",
          activityName: "",
          place: "",
          time: "",
          eventTag: false,
          showError: false
      };
          this.validateForm = this.validateForm.bind(this);
            this.handleSubmit = this.handleSubmit.bind(this);
    }
  validateForm() {
      return this.state.activityName.length > 0 && this.state.place.length > 0;
  }

  handleChange = event => {
      this.setState({
          [event.target.id]: event.target.value
      });
      console.log(this.state);
  }

  handleSubmit = e => {
    e.preventDefault();

    fetch(config.apiPath + "/activity", {
        method: 'POST',
        body: JSON.stringify({
          Description: this.state.description,
          ActivityName: this.state.activityName,
          Place: this.state.place,
          Time: this.state.time,
          EventTag: this.state.eventTag,
          Host: this.state.host
        }),
        headers: {
            'Content-Type': 'application/json'
        }
    }).then((res) => {
        if(!res.ok) {
            throw new Error("Invalid Entries");
        } else if(res.status !== 200) {
            throw new Error("Forbidden");
        }
        return res;
    }).then(res => res.json()).then((res) => {
        //console.log("Token: " + res.token)

        if(typeof (Storage) !== "undefined") {
            localStorage.setItem("apiToken", res.token);
        } else {
            // TODO Code without local storage
        }

        this.props.history.push("/groupmanager");
    }).catch((err) => {
        this.setState({
            errorPrompt: (<Alert bsStyle = "warning"> <strong> Holy guacamole ! </strong>
              Best check yo self, youre not looking too good. </Alert>)
        });
    });

  }

render() {
    return(  <form onSubmit={this.handleSubmit}>
      <FormGroup controlId="errorprompt" bsSize="large">
          {this.errorPrompt}
      </FormGroup>
               	<FormGroup controlId="errorprompt" bsSize="large">
               			{this.state.errorPrompt}
               	</FormGroup>
               	<FormGroup controlId="activityName" bsSize="large">
               			 <FormControl id="activityName" placeholder="activityName" autoFocus="autoFocus" type="text" value={this.state.activityName} onChange={this.handleChange}/>
               	 </FormGroup>
                    <FormGroup controlId="description" bsSize="large">
               			  <FormControl id="description" placeholder="description" autoFocus="autoFocus" type="text" value={this.state.description} onChange={this.handleChange}/>
               			</FormGroup>
                <FormGroup controlId="place" bsSize="large">
                    <FormControl id="place" placeholder="place" autoFocus="autoFocus" type="text" value={this.state.place} onChange={this.handleChange}/>
                </FormGroup>
                <FormGroup controlId="time" bsSize="large">
                  <FormControl id="time" placeholder="time" autoFocus="autoFocus" type="text" value={this.state.time} onChange={this.handleChange}/>
                </FormGroup>
                    <span>Event<Checkbox  id="eventTag" value={this.state.eventTag} onChange={this.handleChange}/></span>
               			<Button id="createBtn" bsStyle="primary" block bsSize="large" disabled={!this.validateForm()} type="submit">
               			  erstellen
               			</Button>
               		  </form>

    )
  }
}
