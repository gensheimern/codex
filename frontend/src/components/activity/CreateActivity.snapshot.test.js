import React from 'react';
import renderer from'react-test-renderer';

import CreateActivity from './CreateActivity';

describe('CreateActivity Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <CreateActivity />
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});