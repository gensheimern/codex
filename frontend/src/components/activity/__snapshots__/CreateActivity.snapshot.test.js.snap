// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateActivity Snapshot renders 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group form-group-lg"
  />
  <div
    className="form-group form-group-lg"
  />
  <div
    className="form-group form-group-lg"
  >
    <input
      autoFocus="autoFocus"
      className="form-control"
      id="activityName"
      onChange={[Function]}
      placeholder="activityName"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group form-group-lg"
  >
    <input
      autoFocus="autoFocus"
      className="form-control"
      id="description"
      onChange={[Function]}
      placeholder="description"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group form-group-lg"
  >
    <input
      autoFocus="autoFocus"
      className="form-control"
      id="place"
      onChange={[Function]}
      placeholder="place"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group form-group-lg"
  >
    <input
      autoFocus="autoFocus"
      className="form-control"
      id="time"
      onChange={[Function]}
      placeholder="time"
      type="text"
      value=""
    />
  </div>
  <span>
    Event
    <div
      className="checkbox"
      style={undefined}
    >
      <label
        title=""
      >
        <input
          disabled={false}
          id="eventTag"
          onChange={[Function]}
          type="checkbox"
          value={false}
        />
      </label>
    </div>
  </span>
  <button
    className="btn btn-lg btn-primary btn-block"
    disabled={true}
    id="createBtn"
    type="submit"
  >
    erstellen
  </button>
</form>
`;
